src/main/java/seedu/address/storage/JsonAdaptedRecipe.java

```diff
@@ -34,10 +36,14 @@
     @JsonCreator
     public JsonAdaptedRecipe(@JsonProperty("name") String name, @JsonProperty("ingredients") String ingredients,
                              @JsonProperty("instructions") String instructions,
+                             @JsonProperty("calorie") String calorie,
                              @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
         this.name = name;
         this.ingredients = ingredients;
         this.instructions = instructions;
+        if (calorie != null) {
+            this.calorie = calorie;
+        }
```

Comment:
since the default value of calorie will be "0", there's no need to check for null alr
can just assign it as normal

Reply:
Why not use the built-in `requireNonNull` method?

Reply:
If im not wrong, its for non optional methods